eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -54,16 +125,32 @@ steps:
         # then pull
         git reset --hard HEAD
         git pull origin master
+        git checkout ${{ parameters.VcpkgVersion }}
 
         ./bootstrap-vcpkg.sh
-    condition: and(succeeded(), not(contains(variables['OSVmImage'], 'macOS')), not(eq(variables['${{ parameters.DependenciesVariableName }}'], '')))
-    displayName: Upgrade vcpkg
+    condition: >-
+      and(
+        succeeded(),
+        ne(variables['VcpkgRestoredFromCache'], 'true'),
+        not(contains(variables['OSVmImage'], 'macOS')),
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], ''))
+      )
+    displayName: Non-MacOS - vcpkg bootstrap
 
   - script: vcpkg version
     displayName: vcpkg version
+    condition: >-
+      and(
+        succeeded(),
+        not(eq(variables['${{ parameters.DependenciesVariableName }}'], ''))
+      )
 
   - script: |
       vcpkg install $(${{ parameters.DependenciesVariableName }})
```

Comment:
Isn't this a step that people might need to run more then once? Or do they pass in all the dependencies?